onnxruntime/core/providers/coreml/builders/impl/argmax_op_builder.cc

```diff
@@ -91,6 +74,19 @@ bool ArgMaxOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderInputPa
     return false;
   }
 
+  // Case where argmax has multiple succeeding nodes(cast node among them) is not supported
+  if (node.GetOutputEdgesCount() > 1) {
+    // Check if Argmax's succeeding nodes contain Cast;If yes, not supported
+    // Otherwise, supported
+    for (auto it = node.OutputEdgesBegin(), end = node.OutputEdgesEnd(); it != end; ++it) {
+      const auto& op_type = it->GetNode().OpType();
+      if (op_type == "Cast") {
+        LOGS(logger, VERBOSE) << "ArgMax has multiple output nodes including Cast";
+        return false;
+      }
+    }
+  }
+
```

Comment:
Comment and code don't quite match. 

If there are multiple output edges what is the desired behavior? Only exit if one of the downstream nodes is a Cast (not quite sure why that matters), or exit whenever there are multiple output edges?

Reply:
It meant to exit only for the case where argmax has both cast and other downstream nodes. Updated the comment.

Regarding the reason, I think since "cast" layer does not exist in coreml model, therefore we now have this special handling only for argmax+cast (by fusing them) so it would be difficult to support another argmax's output nodes ( if this makes sense..